frontend/src/lib/components/Errors/ErrorDisplay.tsx

```diff
@@ -2,66 +2,112 @@ import './ErrorDisplay.scss'
 
-function StackTrace({ frames, showAllFrames }: { frames: StackFrame[]; showAllFrames: boolean }): JSX.Element | null {
-    const displayFrames = showAllFrames ? frames : frames.filter((f) => f.in_app)
+    const initiallyActiveIndex = displayFrames.findIndex((f) => f.in_app) || 0
 
-    const panels = displayFrames.map(({ filename, lineno, colno, function: functionName }, index) => {
+    const panels = displayFrames.map(({ raw_id, source, line, column, resolved_name, lang }, index) => {
+        const frameContext = frameContexts[raw_id]
         return {
             key: index,
             header: (
                 <div className="flex flex-wrap space-x-0.5">
-                    <span>{filename}</span>
-                    {functionName ? (
+                    <span>{source}</span>
+                    {resolved_name ? (
                         <div className="flex space-x-0.5">
                             <span className="text-muted">in</span>
-                            <span>{functionName}</span>
+                            <span>{resolved_name}</span>
                         </div>
                     ) : null}
-                    {lineno && colno ? (
+                    {line ? (
                         <div className="flex space-x-0.5">
                             <span className="text-muted">at line</span>
                             <span>
-                                {lineno}:{colno}
+                                {line}
+                                {column && `:${column}`}
                             </span>
                         </div>
                     ) : null}
                 </div>
             ),
-            content: null,
+            content: frameContext ? <FrameContext context={frameContext} language={getLanguage(lang)} /> : null,
+            className: 'p-0',
         }
     })
 
-    return <LemonCollapse defaultActiveKeys={[]} multiple panels={panels} size="xsmall" />
+    return <LemonCollapse defaultActiveKeys={[initiallyActiveIndex]} multiple panels={panels} size="xsmall" />
+}
+
+function FrameContext({
+    context,
+    language,
+}: {
+    context: ErrorTrackingStackFrameContext
+    language: Language
+}): JSX.Element {
+    const { before, line, after } = context
+    return (
+        <>
+            <FrameContextLine lines={before} language={language} />
+            <FrameContextLine lines={[line]} language={language} highlight />
+            <FrameContextLine lines={after} language={language} />
+        </>
+    )
 }
 
-function ChainedStackTraces({ exceptionList }: { exceptionList: Exception[] }): JSX.Element {
+function FrameContextLine({
+    lines,
+    language,
+    highlight,
+}: {
+    lines: ErrorTrackingStackFrameContextLine[]
+    language: Language
+    highlight?: boolean
+}): JSX.Element {
+    return (
+        <div className={highlight ? 'bg-accent-3000' : 'bg-bg-light'}>
+            {lines.map(({ number, line }) => (
```

Comment:
Might make more sense to sort here, rather than above, actually.

Reply:
I was a little confused by this - where do we sort?